xbmc/Application.cpp

```diff
@@ -2939,6 +2939,42 @@ void CApplication::Stop(int exitCode)
   Sleep(200);
 }
 
+bool CApplication::ResumePlayer(const CFileItemPtr& itemIn) {
+  if (itemIn->IsEmpty())
```

Comment:
It would be nice if you could add the following part above the empty check to respect the startup action for tv and radio.

```c++
int startupAction = CServiceBroker::GetSettings().GetIntValue(CSettings::SETTING_LOOKANDFEEL_STARTUPACTION);
if (startupAction == STARTUP_ACTION_PLAY_TV &&
    startupAction == STARTUP_ACTION_PLAY_RADIO)
{
  CLog::Log(LOGDEBUG, "CApplication::%s startup action to resume TV or radio is enabled", __FUNCTION__);
  return false;
}
```

Reply:
Yes, if you configure 
wakeup setting: use startupaction
then your code will be executed. But for my usecase I do want your code only on cold start but if I left my kodi on my seasons browser and my host goes standby to save payer I would like to resume on browsing instead of getting TV fired on.

I guess at least you would be very unhappy If we don't implement my setting suggestion.

Reply:
Ok, I did miss @FernetMenta suggestion to force the setting "use startupacttion" as the first step

Reply:
You are right, a new setting makes sense if we want to differ between startup / wake-up for PVR. For me the following options should available for the new setting "Wake-up action"

* None
* Resume last played item -> covered by this PR
* Play TV -> should be added to CPVRGUIActions::PlayChannelOnStartup
* Play radio -> should be added to CPVRGUIActions::PlayChannelOnStartup

I would drop the option "use startup action" because it confuses the user or could be difficult in case we add a startup action which could not applied to wake-up.

The default option should be "Resume last played item".

Reply:
Well, there was another unexpected suggestion from @FernetMenta you have to handle startupaction by yourself. Maybe because we don't need a startup action if player is running.